app/components/goodcity/locations-search-overlay.js

```diff
@@ -51,11 +57,25 @@ export default SingletonComponent.extend(SearchMixin, {
     },
 
     loadMoreLocations(pageNo) {
-      const params = this.trimQuery(
-        _.merge({}, this.getSearchQuery(), this.getPaginationQuery(pageNo))
-      );
+      if (this.isValidTextLength()) {
+        if (this.get("presetLocations.length")) {
+          let searchText = this.getSearchQuery().searchText.toLowerCase();
+          return this.get("presetLocations").filter(function(location) {
```

Comment:
We can move filter callback function to separate function.
```
  if (this.get("presetLocations.length")) {
      return this.get("presetLocations").filter(this.isLocationPresent);
  else {
      -----
  } 
```
For `isLocationPresent` method:
```
  isLocationPresent(location) {
     const searchText = this.getSearchQuery().searchText.toLowerCase();
     return (
              location
                .get("name")
                .toLowerCase()
                .indexOf(searchText) > -1
            );
  }
```
This becomes more readable.